### PR TITLE
Obstructions and headless information additions

### PR DESCRIPTION
**Pull request by KarmaKamikaze**
Obstruction has been added to the graph in a dynamic function. The amount of obstruction is controlled with a slider. More thorough order information has been added to the headless information page.

Changes: 
- 200 line buffer for headless simulation order text
- Random obstruction generation
- Slider for the desired number of obstructions
- Update order status to failed upon 24:00 clear
- Added simulation day and delivery success rate to stats
- Added function for generation of stats block
- Refactored dynamic page gen
- Updated index.js
- Reference legend added på visualization page

It will be a decently large amount of work to add good information statistics to the visualization page, so we decided to exclude it for now and mention it in "future work".

### CODE REVIEW

File: node/PublicResources/js/dynamicPageGeneration.js [hunk 1]
```diff
@@ -298,22 +331,11 @@ const generateOptionsHTML = (pageObject) => {
 /**
  * This function generates the headless-simulation html page, which can then be sent as
  * a response to a POST request.
- * @param {Number} graphSize The size of the graphs which will be contained in the divs.
- * @param {String} algorithm The different types of algorithms associated with each graph div.
  * @param {String} graph A deposited div html element in string form, representing the graph container.
+ * @param {String} infoDiv A deposited div html element in string form, representing the information stat block.
  * @returns A string, which contains the generated headless-simulation.html site.
  */
```
Comment (SimonPRasmussen) [inline node/PublicResources/js/dynamicPageGeneration.js:337]: Er Doxygen comments rigtige for ovenstående?

File: node/PublicResources/js/graphHelper.js [hunk 2]
```diff
@@ -358,6 +363,9 @@ class CyGraph {
    * @param {Number} index The index to start from (default: 0)
    */
   animateCourier(path, courier, edges, index = 0) {
+    console.log(
+      `Path = ${path} - Index: ${index} - Path[index]: ${path[index]}`
+    );
```
Comment (damniko) [inline node/PublicResources/js/graphHelper.js]: y dis

File: node/PublicResources/js/graphCore.js [hunk 3]
```diff
@@ -98,9 +98,10 @@ async function startSim() {
       DISTANCE_PER_TICK, // courier movement speed
       graphSettings["order-frequency"], // order rate (pr restaurant)
       graphSettings["idle-zones"], // use idle zones
-      true, // headless traversal
+      graphDivs[i].classList.contains("headless"), // headless traversal
       graphSettings["courier-frequency"], // max number of couriers
-      tickSpeed // ms per tick
+      tickSpeed, // ms per tick
+      graphSettings["obstruction-level"]
```
Comment (SimonPRasmussen) [inline node/PublicResources/js/graphCore.js:104]: Comment på det her ligesom de andre?

File: node/PublicResources/js/heatGeneration.js [hunk 4]
```diff
@@ -192,3 +192,38 @@ function findNodesInRadius(cyGraph, startNode, radius) {
...
+    obstructionEdge.data("obstructions", (obstructionVal * 2) / 3); // Scale edge's obstructions down to original obstruction value
+    cyGraph.calculateWeight(obstructionEdge.id()); // Update the weight property of the edge
+    obstructionEdge.removeClass(eleType.obstructions); // Remove class marking edge as obstruction edge
+  }
+  cyGraph.obstructions = []; // Empty obstructions array
+
+  let edges = cyGraph.graph.edges();
+  for (let i = 0; i < cyGraph.obstructionLevel; i++) {
+    // Create obstructions on random edges
+    let edge = edges[Math.floor(Math.random() * edges.length)]; // Get random edge
```
Comment (damniko) [inline node/PublicResources/js/heatGeneration.js]: kunne bruge cyGraph.getRandomInt(edges.length) for clarity (når nu vi allerede har lavet den funktion)

**Review by damniko — APPROVED** [review]
noice

File: node/PublicResources/js/dynamicPageGeneration.js [hunk 5]
```diff
@@ -251,7 +260,11 @@ const generateOptionsHTML = (pageObject) => {
             oninput="this.nextElementSibling.value = this.value"
           />
           <output>0.25</output>
-          <p>Ticks per second:</p>
+          <br>
+          <div class="tooltip">Ticks per second:
+            <span class="tooltiptext">10 or below for visualization - above 10 for headless.</span>
```
Comment (Sarmisuper) [inline node/PublicResources/js/dynamicPageGeneration.js]: Hvorfor er det ticks per second der styrer om det er headless eller ej? Jeg troede vi havde en anden måde at definere det på?

**Review by Sarmisuper — APPROVED** [review]

File: node/PublicResources/js/graphHelper.js (hunk 2 above)
Comment (mirakst) [reply node/PublicResources/js/graphHelper.js]: fixed

File: node/PublicResources/js/heatGeneration.js (hunk 4 above)
Comment (mirakst) [reply node/PublicResources/js/heatGeneration.js]: fixed

**Review by SimonPRasmussen — APPROVED** [review]
Looks very nice 👌

**Review by KasperHenningsen — APPROVED** [review]

**Pull request closed without merging**